web/src/features/prompts/server/handlers/promptNameHandler.ts

```diff
@@ -0,0 +1,26 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { getPromptByName } from "@/src/features/prompts/server/actions/getPromptByName";
+import { GetPromptByNameSchema } from "@/src/features/prompts/server/utils/validation";
+import { withMiddlewares } from "@/src/server/utils/withMiddlewares";
+import { MethodNotAllowedError } from "@langfuse/shared";
+
+import { authorizePromptRequestOrThrow } from "../utils/authorizePromptRequest";
+
+const handler = async (req: NextApiRequest, res: NextApiResponse) => {
+  const authCheck = await authorizePromptRequestOrThrow(req);
+  if (req.method !== "GET") throw new MethodNotAllowedError();
+
+  const { promptName, version, label } = GetPromptByNameSchema.parse(req.query);
```

Comment:
Please correct me if i am wrong: we decided to drop the promptId requirement here and go for a different api ressource for that goinf forward. Correct?